cmd/kueuectl/app/list/list_pods.go

```diff
@@ -0,0 +1,237 @@
+	return &PodOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewPodCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewPodOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:                   "pods --for [<type>[.<api-group>]/]<name>",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"po"},
+		Short:                 "List Pods belong to a Job Kind",
+		Long:                  podLong,
+		Example:               podExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter))
+			cobra.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	addAllNamespacesFlagVar(cmd, &o.AllNamespaces)
+	addFieldSelectorFlagVar(cmd, &o.FieldSelector)
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	addForObjectFilterFlagVar(cmd, &o.UserSpecifiedForObject)
+
+	_ = cmd.MarkFlagRequired("for")
+
+	return cmd
+}
+
+// Complete takes the command arguments and infers any remaining options.
+func (o *PodOptions) Complete(clientGetter util.ClientGetter) error {
+	var err error
+
+	o.Limit, err = listRequestLimit()
+	if err != nil {
+		return err
+	}
+	o.Namespace, _, err = clientGetter.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.Clientset, err = clientGetter.K8sClientSet()
+	if err != nil {
+		return err
+	}
+
+	// parse --for flag
+	if o.UserSpecifiedForObject != "" {
+		mapper, err := clientGetter.ToRESTMapper()
+		if err != nil {
+			return err
+		}
+		var found bool
+		o.ForGVK, o.ForName, found, err = decodeResourceTypeName(mapper, o.UserSpecifiedForObject)
+		if err != nil {
+			return err
+		}
+		if !found {
+			return fmt.Errorf("invalid value '%s' used in --for flag; value must be in the format [TYPE[.API-GROUP]/]NAME", o.UserSpecifiedForObject)
+		}
+	}
+
+	return nil
+}
+
+func (o *PodOptions) ToPrinter(headers bool) (printers.ResourcePrinterFunc, error) {
+	if !o.PrintFlags.OutputFlagSpecified() {
+		printer := newPodTablePrinter().
+			WithNamespace(o.AllNamespaces).
+			WithNoHeaders(headers)
+		return printer.PrintObj, nil
+	}
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return nil, err
+	}
+
+	return printer.PrintObj, nil
+}
+
+// Run prints the pods for a specific Job
+func (o *PodOptions) Run(ctx context.Context) error {
+	err := o.listPods(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// listPods lists the pods based on the given --for object
+func (o *PodOptions) listPods(ctx context.Context) error {
+	var totalCount int
+
+	namespace := o.Namespace
+	if o.AllNamespaces {
+		namespace = ""
+	}
+
+	opts := metav1.ListOptions{
+		LabelSelector: o.LabelSelector,
+		FieldSelector: o.FieldSelector,
+		Limit:         o.Limit,
+	}
+
+	tabWriter := printers.GetNewTabWriter(o.Out)
+	for {
+		podList, err := o.Clientset.CoreV1().Pods(namespace).List(ctx, opts)
```

Comment:
This is not going to work well.... a cluster can have 10s or 100s of thousands of Pods. A list call on that without any significant label selectors could cause problems to the apiserver if overused.

We will have to specialize this further.

For example, Jobs have a known label selector. Once you have identified that the query is for a job, you retrieve the job object, get the label selector and include that in the List call.

Perhaps the job framework interface can be extended to have a method `PodLabelSelector`, so that it can be done a bit more generically.

For

Reply:
Yeah, I like to generalise than specializing this for each integrations/Kind. Can I get more info `JobFramework` route?. Apologies, I'm just going through the code.

Reply:
Not sure I'm understanding it right. So we need to implement a method `PodLabelSelector` in all Job kinds which gives us the new label value relates between the actual integration/Kind and the Pods?

Reply:
it should give you the label selector that would match all the pods for that job.

Reply:
Sorry was out for a while last week.

Can you elaborate on those labels part please? Do we just need to add a new receiver function and return the selectorLabels (?) or introduce any new labels.

Reply:
return the selectorLabels that most jobs already use to identify their own pods. For example, for Job, it is something like `batch.kubernetes.io/job-name=foo`

Reply:
okay, I just updated https://github.com/kubernetes-sigs/kueue/pull/2280/commits/a6663de6c6134d456338b8220846873dde740441 to list pods based on standard labels by job kind. Hope you were referring to the same, still I'm not sure if these labels will be consistent for all the job's pods.